spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -110,6 +120,27 @@ object DynamicInterpreter {
     optimizerTimeTracker.cleanupWorkaroundDf(conformedDf)
   }
 
+  private def checkMutabilityNotViolated(inputDf: DataFrame, steps: List[ConformanceRule])
```

Comment:
```suggestion
  private def validateDropRules(inputDf: DataFrame, steps: List[ConformanceRule])
```

Reply:
The method name is one thing to consider; I'd maybe only pass `inputSchema: StructType`, because the DataFrame is actually not needed at all.